sycl/test/extensions/properties/new_properties_open_question.cpp

```diff
@@ -0,0 +1,185 @@
+  static constexpr int value = N;
+};
+struct other_prop : named_property_base<other_prop> {};
+
+void bar() {
+  foo(properties{prop<42>{}});
+  foo(properties{prop<42>{}, other_prop{}});
+}
+} // namespace approach_one
+
+namespace approach_two {
+// Keep providing `inline constexpr` "shortcuts".
+template <int N>
+struct prop_property : named_property_base<prop_property<N>, prop_key> {
+  static constexpr int value = N;
+};
+struct other_prop_property : named_property_base<other_prop_property> {};
+
+template <int N> inline constexpr prop_property<N> prop{};
+inline constexpr other_prop_property other_prop{};
+
+void bar() {
+  // Still need `properties` here.
+  foo(properties{prop<42>});
+  foo(properties{prop<42>, other_prop});
+}
+} // namespace approach_two
+namespace approach_three {
+// Use operator+ to create `properties` property list from individual
+// properties.
+template <int N>
+struct prop_property : named_property_base<prop_property<N>, prop_key> {
+  static constexpr int value = N;
+};
+struct other_prop_property : named_property_base<other_prop_property> {};
+
+template <int N> inline constexpr prop_property<N> prop{};
+inline constexpr other_prop_property other_prop{};
+
+void bar() {
+  // Unary `+` has a bit of hacky feeling...
+  foo(+prop<42>);
+  // More than one property in a property list looks very natural.
+  // Alternatively, that can be `operator|` but it has no unary version.
+  foo(prop<42> + other_prop);
+}
+} // namespace approach_three
+
+namespace approach_four {
+// "Duck-typing" - make individual properties behave almost like a property
+// list.
+
+// This will be part of the implementation's `detail::property_base` but I'm
+// keeping it in the test for now until the future direction is chosen.
+template <typename property_t, typename property_key_t = property_t>
+struct adjusted_property_base
+    : public named_property_base<property_t, property_key_t> {
+  template <typename T> static constexpr bool has_property() {
+    return std::is_same_v<T, property_key_t>;
+  }
+
+  // Technically it should be two version static/non-static depending on
+  // `std::is_empty_v<property_t>`, skipped here for brevity.
+  template <typename T,
+            typename = std::enable_if_t<std::is_same_v<T, property_key_t>>>
+  static constexpr property_t get_property() {
+    return property_t{};
+  }
+};
+
+template <int N>
+struct prop_property : adjusted_property_base<prop_property<N>, prop_key> {
+  static constexpr int value = N;
+};
+struct other_prop_property : adjusted_property_base<other_prop_property> {};
+
+template <int N> inline constexpr prop_property<N> prop{};
+inline constexpr other_prop_property other_prop{};
+
+void bar() {
+  // "Duck-typing" here.
+  foo(prop<42>);
+
+  // Now nothing prevents us from using `operator|` here that someone might
+  // found more natural. Not doing for simplicity for now.
+  foo(prop<42> + other_prop);
+}
```

Comment:
Another way that we could achieve this syntax without the duck-typing would be to define two overloads of every function that is supposed to take properties:
```c++
template <typename Property> auto foo(Property prop);
template <typename PropertyList> auto foo(PropList props = {});
```

...but there would be drawbacks to that, like the fact that `foo<prop<42>>` would be a different function to `foo<properties<prop<42>>>`.  That said, I think duck-typing and @rolandschulz's suggestion that we handle this via adjusted constraints might both have this same issue.

Would this be good enough?  People designing property interfaces would have to be aware of this potential issue, but we might just be able to disable the single-property overload on a case-by-case basis.  I think it would only matter if somebody wanted to use a `static` variable inside of a function accepting a property list, and I can't immediately think of a use-case for that.  And there's a straightforward workaround (as @rolandschulz pointed out offline): if you want to use this pattern, `foo(property)` should just call `foo(properties(property))`.

Providing the extra overload seems nicer to me than supporting unary `+`.

Reply:
> Providing the extra overload seems nicer to me than supporting unary `+`.

I agree.  Would callers be able to pass an initializer list to specify a list of properties like:

```
foo({prop1, prop2});
```

I think that would be nicer than:

```
foo(properties{prop1, prop2});
```

Reply:
> Would callers be able to pass an initializer list to specify a list of properties like:

I might be wrong, but I think we'll always hit problems with this sort of thing because the property list type can't be deduced.

We can make `foo((prop1, prop2))` work by overloading the `,(property)` operator, but I don't know if that's a good idea in general.  I've never seen `,` overloaded in the wild.